packages/ethereum-eip1559-fee-provider/lib/EthereumEIP1559FeeProvider.ts

```diff
@@ -0,0 +1,188 @@
+import { JsonRpcProvider } from '@liquality/jsonrpc-provider'
+import { FeeProvider, FeeDetails, BigNumber, FeeHistory } from '@liquality/types'
+import { gwei, toGwei, hexToNumber, numberToHex } from '@liquality/ethereum-utils'
+
+type FeeOptions = {
+  slowMultiplier?: number
+  averageMultiplier?: number
+  fastMultiplier?: number
+  maxFeePerGasThreshold?: number
+}
+
+interface FeeEstimationResult {
+  maxFeePerGas: BigNumber
+  maxPriorityFeePerGas: BigNumber
+  baseFeePerGas?: BigNumber
+}
+
+const MAX_GAS_FAST = gwei(1500)
+// How many blocks to consider for priority fee estimation
+const FEE_HISTORY_BLOCKS = 10
+// Which percentile of effective priority fees to include
+const FEE_HISTORY_PERCENTILE = 5
+// Which base fee to trigger priority fee estimation at
+const PRIORITY_FEE_ESTIMATION_TRIGGER = gwei(100)
+// Returned if above trigger is not met
+const DEFAULT_PRIORITY_FEE = gwei(3)
+// In case something goes wrong fall back to this estimate
+export const FALLBACK_ESTIMATE: FeeEstimationResult = {
+  maxFeePerGas: gwei(20),
+  maxPriorityFeePerGas: DEFAULT_PRIORITY_FEE,
+  baseFeePerGas: undefined
+}
+const PRIORITY_FEE_INCREASE_BOUNDARY = 200 // %
+
+export default class EthereumEIP1559FeeProvider extends JsonRpcProvider implements FeeProvider {
+  _slowMultiplier: number
+  _averageMultiplier: number
+  _fastMultiplier: number
+  _maxFeePerGasThreshold: number
+
+  constructor(options: { uri: string; username?: string; password?: string }, opts: FeeOptions = {}) {
+    super(options.uri, options.username, options.password)
+
+    const { slowMultiplier = 1, averageMultiplier = 1.25, fastMultiplier = 1.4, maxFeePerGasThreshold = 1000 } = opts
+    this._slowMultiplier = slowMultiplier
```

Comment:
are the multipliers needed anymore?
I mean i thought the EIP1559 logic itself can give good recommendations
the main use case of this multiplier was that the estimation come from node for legacy transaction was not very accurate

Reply:
Started using https://github.com/rainbow-me/fee-suggestions and it seems to be handling this well with surge in fees.